src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs

```diff
@@ -182,19 +186,34 @@ public void Initialize()
             // Warn the user that execution will wait for debugger attach.
             var processId = this.dataCollectionLauncher.LaunchDataCollector(null, this.GetCommandLineArguments(port));
 
-            var dataCollectorDebugEnabled = Environment.GetEnvironmentVariable("VSTEST_DATACOLLECTOR_DEBUG");
-            if (!string.IsNullOrEmpty(dataCollectorDebugEnabled) && dataCollectorDebugEnabled.Equals("1", StringComparison.Ordinal))
+            ChangeConnectionTimeoutIfRequired(processId);
+            var connected = this.dataCollectionRequestSender.WaitForRequestHandlerConnection(this.connectionTimeout);
+            if (connected == false)
+            {
+                throw new TestPlatformException(string.Format(CultureInfo.CurrentUICulture, CrossPlatEngineResources.FailedToConnectDataCollector));
+            }
+        }
+
+        private void ChangeConnectionTimeoutIfRequired(int processId)
```

Comment:
There must be a similar utility method for ProxyExecutionManager, can you please check it, & merge the two.